src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -262,16 +262,16 @@ private function fixPhpDoc(Tokens $tokens, int $index, array $uses, string $name
     {
         $phpDoc = $tokens[$index];
         $phpDocContent = $phpDoc->getContent();
-        Preg::matchAll('#@([^\s]+)\s+([^\s]+)#', $phpDocContent, $matches);
+        Preg::matchAll('#@([^\s]+)(\s+)((?:[a-zA-Z0-9_\\\\]+)+)#', $phpDocContent, $matches);
```

Comment:
please use here matching using `TypeExpression::REGEX_TYPES` like in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.42.0/src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php#L523, `Preg::match` should be then enough

Reply:
No, it's a bug fix only, using full regex would change the behaviour more - support was added for simple FQCNs, not for complex types. We can improve it later.

Reply:
IMHO non-grammar based fixer should never be added as the regex would match `int` of `int-xxx` wrongly. So I prefer to revert the original PR and fix all the issues properly.

Also, https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/5620#discussion_r1412877786 is broken with simple types:
```diff
      /**
-     * @return int
+     * @return \int
       */
```

Reply:
Also this is broken:
```diff
-                } catch (\Exception $e) {
+                } catch (Exception $e) {
```
with this config:
```
        'fully_qualified_strict_types' => [
            'leading_backslash_in_global_namespace' => true,
        ],
```

Reply:
TBH, I do not see an reason why to to replace the questionable regex with the proper type regex immediately.

Reply:
Because as far as I understand it would match complex types, which means it would require improved logic for processing `$matches`, and I did not have time to work on this at this point. I wanted to provide the simplest solution which fixes the problem. I have holidays until January 3rd 😉.

Reply:
The LOC needed is close to a few, I should probably keep you working in this, just please include all tests I added in #7623